pkg/transform/kubernetes/kubernetes.go

```diff
@@ -106,6 +106,16 @@ func (t *Transformer) CreateDeployments(o *object.Service) ([]runtime.Object, er
 		},
 	}
 
+	// Set replica value only if it is given
+	if o.Replicas != nil {
+		// and is positive number
+		if *o.Replicas >= 0 {
+			d.Spec.Replicas = o.Replicas
+		} else {
+			return nil, fmt.Errorf("Replica count is negative in service: %q", o.Name)
+		}
+	}
```

Comment:
May be we could use [object.Validate](https://github.com/redhat-developer/opencompose/blob/51b38c48fc8af3bb815e842629b700bfff5c6dd6/pkg/object/object.go#L50) for this.